src/tum_dlr_automl_for_eo/utils/custom_metrics.py

```diff
@@ -1,4 +1,102 @@
 import numpy as np
+from random import sample
+
+
+def get_neighbors(architecture):
+    """
+    @param architecture: 1 binary encoded architecture np.array
+    @return: 5 neighbors of given arc. with 1 Hamming Distance
+    """
+    # get 5 random indices to create 5 neighbors
+    random_neighbor_indices = sample(range(0, len(architecture) - 1), 5)
+    neighbors = []
+    for index in random_neighbor_indices:
+        # Change architecture to keep Hamming Distance at 1
+        if architecture[index] == 0:
+            curr_architecture = architecture.copy()
+            curr_architecture[index] = 1
+        else:
+            curr_architecture = architecture.copy()
+            curr_architecture[index] = 0
+        neighbors.append(curr_architecture)
+    return neighbors
+
+
+def perform_bils_algorithm(architectures, acc_list, starting_point):
+    """
+
+    @param architectures: np.array [(Number of Models) x (length of binary encoded architecture)] binary encoded
+    architectures
+    @param acc_list: np.array [Number of Models] Final Accuracies of Models @param starting_point: int
+    Starting model to search
+    @return: list [found architecture index, found architecture's accuracy]
+             in case of no improvement returns given architecture info.
+    """
+    MAX_NUMBER_OF_ITERATIONS = 50
```

Comment:
What do you think about making this a parameter?

Reply:
i think this is something to be decided as a result of cost analysis and will be a fixed parameter, but ideally we can

Reply:
This is something I will modify